client/src/components/Markdown/Elements/HistoryDatasetAsTable.vue

```diff
@@ -0,0 +1,126 @@
+<script setup lang="ts">
+import { computed } from "vue";
+
+import { UrlDataProvider } from "@/components/providers/UrlDataProvider.js";
+
+import LoadingSpan from "@/components/LoadingSpan.vue";
+
+interface HistoryDatasetAsTableProps {
+    historyDatasetId: string;
+    compact: boolean;
+    showColumnHeaders: boolean;
+    title?: string;
+    footer?: string;
+}
+
+const props = withDefaults(defineProps<HistoryDatasetAsTableProps>(), {
+    compact: false,
+    showColumnHeaders: true,
+    title: null,
+    footer: null,
```

Comment:
```suggestion
    title: undefined,
    footer: undefined,
```
Ether this or declare the prop as `title?: string | null;`.

If I'm not mistaken, [the default tsconfig we are extending](https://github.com/jmchilton/galaxy/blob/06c48dcd6e430327b197d4c61695a58f7f47b7cc/client/tsconfig.json#L2) has `"strictNullChecks": true`.

The same applies in general for other optional props with defaults in this PR.

Reply:
There are things around strings and nulls in Typescript and these properties that just don't ever click for me. It will look fine and then not compile. I will try out though - it makes sense on in theory to me.